ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -0,0 +1,640 @@
+
+        # now = time.time()
+        results = self._rdfGraph.query(query)
+        # Depending on the cardinality this query can return multiple rows
+        # per annotations.  Here we reduce it to a list of unique annotations
+        # URIs
+        uniqueAnnotations = sets.Set()
+        for row in results:
+            uniqueAnnotations.add("<{}>".format(row['s'].toPython()))
+
+        annotations = AnnotationFactory(self._rdfGraph,
+                                        uniqueAnnotations).annotations()
+
+        # responseTime = time.time()-now
+        # if responseTime > self.RESPONSETIME_LOGGING_THRESHOLD:
+        #     print('queryLabels', responseTime)
+        #     print('len(annotations)', len(annotations))
+
+        return annotations
+
+    def _formatQuery(self, location=None, drug=None, disease=None):
+        """
+        Generate a formatted sparql query with appropriate filters
+        """
+        query = self._searchQuery
+        if location is None and drug is None and disease is None:
+            raise exceptions.NotImplementedException(
+               "At least one of [location, drug, disease] must be specified")
+        filters = []
+
+        if location and isinstance(location, basestring):
+            filters.append('regex(?location_label, "{}")'.format(location))
+        if drug and isinstance(drug, basestring):
+            filters.append('regex(?drug_label, "{}")'.format(drug))
+        if disease and isinstance(disease, basestring):
+            filters.append('regex(?disease_label, "{}")'.format(disease))
+
+        locationClause = ""
+        if isinstance(location, dict):
+            locations = []
+            for id in location['ids']:
+                    locations.append('?location = <{}> '.format
+                                     (id['database'] + id['identifier']))
+            locationClause = "({})".format(" || ".join(locations))
+            filters.append(locationClause)
+            locationClause = "?l  faldo:location ?location .\n"
+
+        if isinstance(drug, dict):
+            drugs = []
+            for id in drug['ids']:
+                    drugs.append('?drug = <{}> '.format
+                                 (id['database'] + id['identifier']))
+            drugsClause = "({})".format(" || ".join(drugs))
+
+            filters.append(drugsClause)
+
+        if isinstance(disease, dict):
+            diseases = []
+            for id in disease['ids']:
+                    diseases.append('?disease = <{}> '.format
+                                    (id['database'] + id['identifier']))
+            diseasesClause = "({})".format(" || ".join(diseases))
+            filters.append(diseasesClause)
+
+        filter = "FILTER ({})".format(' && '.join(filters))
+        query = query.replace("%FILTER%", filter)
+        query = query.replace("%LOCATION%", locationClause)
+        return query
+
+# flake8: noqa
+    def testData(self):
+        return """
```

Comment:
I'm not a huge fan of this --- what function is it serving? We can certainly add the data to the tests data in tests/data/[somewhere], just embedding it isn't good practice.